test/e2e/adminapi_redeployvm.go

```diff
@@ -50,7 +51,8 @@ var _ = Describe("[Admin API] VM redeploy action", func() {
 			events, err := clients.Kubernetes.EventsV1().Events("default").List(ctx, metav1.ListOptions{})
 
 			if err != nil {
-				return false, err
+				log.Warn(fmt.Sprintf("Got unexpected error %v while trying to fetch events, retrying...", err))
```

Comment:
I've noticed the trend is we can get a "node returned to ready state", but typically this flake fails to match on the `event.Reason == "Rebooted"` and/or `event.Reason == "NodeNotReady"`.
+1 for more logging

Reply:
Link of that for documentation purposes: https://msazure.visualstudio.com/AzureRedHatOpenShift/_build/results?buildId=52713570&view=logs&j=10f66467-59e5-53bc-0c8a-42cc89770de4&t=3bbe7e73-d181-5aca-4b77-91acc0ef0f4a&l=1161

Reply:
I added some extra logic, it looks like the Node actually never reboots, even though the request through the azure SDK returns success and returns no errors: https://msazure.visualstudio.com/AzureRedHatOpenShift/_build/results?buildId=52920649&view=logs&j=10f66467-59e5-53bc-0c8a-42cc89770de4&t=3bbe7e73-d181-5aca-4b77-91acc0ef0f4a&l=1147 I'm gonna try to reproduce this with some minimal code and see if the azure SDK version we're on is just flaking out for some reason or dropping errors...